posthog/temporal/workflows/logger.py

```diff
@@ -0,0 +1,272 @@
+import asyncio
+import json
+import logging
+
+import aiokafka
+import structlog
+import temporalio.activity
+import temporalio.workflow
+from django.conf import settings
+from structlog.processors import EventRenamer
+from structlog.typing import FilteringBoundLogger
+
+from posthog.kafka_client.topics import KAFKA_LOG_ENTRIES
+
+
+async def bind_batch_exports_logger(team_id: int, destination: str | None = None) -> FilteringBoundLogger:
+    """Return a bound logger for BatchExports."""
+    if not structlog.is_configured():
+        await configure_logger()
```

Comment:
Also, I guess leaving these promises dangling is cool in Python? (Related to the other async question I asked about loggers, I guess.)

```
return (listen_task, worker_shutdown_handler_task)
```

Reply:
Aah, good catch, yeah, we should keep a reference to these tasks as they could be gc'd otherwise.

Reply:
> Couldn't this be called concurrently while another task is waiting on configure_logger to complete?

Yeah, but `configure_logger` is actually all sync. I probably got confused because it's spawning an async task, but not async itself. So, we can just make this function sync and get rid of any potential race conditions.

Reply:
I've kept a `BACKGROUND_LOGGER_TASKS` set to hold references to these tasks. This way, they won't be garbage collected and disappear.